x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/analytics_list.tsx

```diff
@@ -119,6 +124,18 @@ export const DataFrameAnalyticsList: FC<Props> = ({
     blockRefresh
   );
 
+  const selectedId = getSelectedJobIdFromUrl(window.location.href);
+  useEffect(() => {
+    if (urlFilterIdCleared === false && analytics.length > 0) {
+      if (selectedId !== undefined) {
+        setUrlFilterIdCleared(true);
+        setQueryText(selectedId);
+        const selectedIdQuery: Query = EuiSearchBar.Query.parse(selectedId);
+        onQueryChange({ query: selectedIdQuery, error: undefined });
+      }
+    }
+  }, [urlFilterIdCleared, analytics]);
```

Comment:
Could the `useEffect` dependencies be `selectedId`? I'm not sure why you'd need to re-run it when `analytics` changes. Might be missing context here.